frontend/packages/shipwright-plugin/src/components/build-tabsection/BuildOverview.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { ResourceLink, SidebarSectionHeading } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s/k8s';
+import { OverviewItem } from '@console/shared';
+import { BUILDRUN_TO_RESOURCE_MAP_LABEL } from '../../const';
+import { BuildModel } from '../../models';
+import { Build, BuildRun } from '../../types';
+import { byCreationTime } from '../../utils';
+import BuildRunItem from './BuildRunItem';
+import StartBuildButton from './StartBuildButton';
+import TriggerLastBuildButton from './TriggerLastBuildButton';
+
+const MAX_VISIBLE = 3;
+
+type BuildsOverviewProps = {
+  item: OverviewItem & {
+    builds?: Build[];
+    buildRuns?: BuildRun[];
+  };
+};
+
+const BuildsOverview: React.FC<BuildsOverviewProps> = ({ item: { builds, buildRuns, obj } }) => {
+  const { t } = useTranslation();
+  const resourceLabel = obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL];
+  const buildRunsforResource = resourceLabel
+    ? buildRuns.filter((buildRun) => {
+        return resourceLabel === buildRun.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL];
+      })
+    : [];
+
+  const buildsForResource = builds.filter((build) => {
+    return (
+      build.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL] ===
+      obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
+    );
+  });
+
+  if (!buildsForResource || !buildsForResource.length) {
+    return null;
+  }
+
+  return (
+    <>
+      <SidebarSectionHeading text={t('shipwright-plugin~BuildRuns')}>
+        {buildRunsforResource.length > MAX_VISIBLE && (
+          <Link
+            className="sidebar__section-view-all"
+            to={`/k8s/ns/${obj.metadata?.namespace}/shipwright.io~v1alpha1~BuildRun?labels=${BUILDRUN_TO_RESOURCE_MAP_LABEL}=${obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]}`}
+          >
+            {t('shipwright-plugin~View all {{buildLength}}', {
+              buildLength: buildRunsforResource.length,
+            })}
+          </Link>
```

Comment:
it would have been nice if would have kept the version, group, kind as model or const so that in the future if the version gets updated would have been one place change.

or can't we use `${resourcePath(kind, name, namespace)}/logs`



```suggestion
          <Link
            className="sidebar__section-view-all"
            to={`${resourcePath(referenceForModel(BuildRunModel), name, namespace)}?labels=${BUILDRUN_TO_RESOURCE_MAP_LABEL}=${obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]}`}
          >
            {t('shipwright-plugin~View all {{buildLength}}', {
              buildLength: buildRunsforResource.length,
            })}
          </Link>
```